go/src/koding/klientctl/endpoint/machine/mount.go

```diff
@@ -201,25 +206,80 @@ func (c *Client) Umount(options *UmountOptions) (err error) {
 		return errors.New("invalid nil options")
 	}
 
-	// TODO: ask user to confirm unmounting.
-	fmt.Fprintf(os.Stdout, "Unmounting %s...\n", options.Identifier)
+	// Get all mounts if all option is set.
+	identifiers := options.Identifiers
+	if options.All {
+		mids, err := c.allMounts()
+		if err != nil {
+			return err
+		}
+		identifiers = mids
+	}
 
-	// Remove mount.
-	umountReq := &machinegroup.UmountRequest{
-		Identifier: options.Identifier,
+	// If there are no mounts we return nil.
+	if len(identifiers) == 0 {
+		fmt.Fprintf(os.Stdout, "There is nothing to unmount\n")
+		return nil
 	}
-	var umountRes machinegroup.UmountResponse
 
-	if err := c.klient().Call("machine.umount", umountReq, &umountRes); err != nil {
-		return err
+	// Do not be too verbose.
+	promptSuffix := strings.Join(identifiers, ", ")
+	if n := len(identifiers); n > 2 {
+		promptSuffix = strconv.Itoa(n) + " mounts"
 	}
 
-	fmt.Fprintf(os.Stdout, "Successfully unmounted %s (ID: %s)\n",
-		umountRes.Mount, umountRes.MountID)
+	fmt.Fprintf(os.Stdout, "Unmounting %s...\n", promptSuffix)
+
+	var yn = "yes"
+	// TODO(ppknap): remove second condition when atom package implements "force" flag.
+	if !options.Force && len(identifiers) > 1 {
+		yn, err = helper.Ask("This operation will remove all cached data. Do you want to continue [y/N]: ")
+		if err != nil {
+			return err
+		}
+	}
+
+	if yn = strings.ToLower(yn); yn != "yes" && yn != "y" {
+		return errors.New("aborted by user")
+	}
+
+	// Remove mounts.
+	for _, identifier := range identifiers {
+		umountReq := &machinegroup.UmountRequest{
+			Identifier: identifier,
+		}
+		var umountRes machinegroup.UmountResponse
+
+		if err := c.klient().Call("machine.umount", umountReq, &umountRes); err != nil {
```

Comment:
What about making it best-effort attempt at unmounting all mounts? So if single unmount fails we keep going unmounting test of them.